src/jdmtool/data_card/detect.py

```diff
@@ -49,47 +49,61 @@ def _open_usb_device(usbdev: USBDevice):
 
 
 SKYBOUND_VID_PID = (0x0E39, 0x1250)
-GARMIN_UNINITIALIZED_VID_PID = (0x091E, 0x0500)
-GARMIN_VID_PID = (0x091E, 0x1300)
-# If you reset the Garmin Programmer's EEPROM, you end up with a Cypress EZ-USB FX2
-FX2_VID_PID = (0x04B4, 0x8613)
+GARMIN_UNINITIALIZED_VID_PID = {
+    (0x091E, 0x0500), # "newer" 010-10579-20 
+    (0x091E, 0x0300), # "older" 011-01277-00
+    (0x04B4, 0x8613), # Cypress EZ-USB FX2 (if EEPROM is reset)
+}
 
+GARMIN_VID_PID = (0x091E, 0x1300)
 
 @contextmanager
 def open_programming_device() -> Generator[ProgrammingDevice, None, None]:
     with USBContext() as usbcontext:
         dev_cls: type[ProgrammingDevice] | None = None
+        read_ep: int | None = None
+        write_ep: int | None = None
         for usbdev in usbcontext.getDeviceIterator():
-            vid_pid: tuple[int, int] = (usbdev.getVendorID(), usbdev.getProductID())
+            # Debug: log each device found
+            vid = usbdev.getVendorID()
+            pid = usbdev.getProductID()
+            vid_pid: tuple[int, int] = (vid, pid)
+            print(f"Identified USB device at {usbdev}: Vendor ID=0x{vid:04X}, Product ID=0x{pid:04X}")
```

Comment:
This is a bit too much output. (Could add a "verbose" option, or maybe use the "logging" module... but I'd leave that for a different PR.)